reducers/plugins/index.ts

```diff
@@ -175,6 +176,7 @@ const initialComponents: PluginsState['components'] = {
     FilePreview: [],
     LinkTooltip: [],
     MainMenu: [],
+    ChannelHeaderButton: [],
```

Comment:
In https://github.com/mattermost/mattermost-webapp/pull/9273, I added these empty arrays to initialize the store, mainly to make the stored more strongly typed. I missed adding the `ChannelHeaderButton` components, which exposed the bug of component cleanup when we remove a component.